front/app/containers/Admin/projects/project/timeline/edit.tsx

```diff
@@ -415,132 +294,100 @@ class AdminProjectTimelineEdit extends PureComponent<
-                  withCTAButton
-                />
-                <Error apiErrors={errors && errors.description_multiloc} />
-              </SectionField>
-
-              <SectionField>
-                <FileUploader
-                  id="project-timeline-edit-form-file-uploader"
-                  onFileAdd={this.handlePhaseFileOnAdd}
-                  onFileRemove={this.handlePhaseFileOnRemove}
-                  files={inStatePhaseFiles}
-                  apiErrors={errors}
-                />
-              </SectionField>
-
-              {errors && errors.project && (
-                <SectionField>
-                  <Error apiErrors={errors.project} />
-                </SectionField>
-              )}
-              {errors && errors.base && (
-                <SectionField>
-                  <Error apiErrors={errors.base} />
-                </SectionField>
-              )}
-            </Section>
-
-            <SubmitWrapper
-              loading={processing}
-              status={submitState}
-              messages={{
-                buttonSave: messages.saveLabel,
-                buttonSuccess: messages.saveSuccessLabel,
-                messageError: messages.saveErrorMessage,
-                messageSuccess: messages.saveSuccessMessage,
-              }}
+  const phaseAttrs = !isNilOrError(phase)
+    ? { ...phase.attributes, ...attributeDiff }
+    : { ...attributeDiff };
+  const startDate = getStartDate();
+  const endDate = phaseAttrs.end_at ? moment(phaseAttrs.end_at) : null;
+
+  return (
+    <>
+      <SectionTitle>
+        {phase && <FormattedMessage {...messages.editPhaseTitle} />}
+        {!phase && <FormattedMessage {...messages.newPhaseTitle} />}
+      </SectionTitle>
+
+      <PhaseForm onSubmit={handleOnSubmit}>
+        <Section>
+          <SectionField>
+            <InputMultilocWithLocaleSwitcher
+              id="title"
+              label={<FormattedMessage {...messages.titleLabel} />}
+              type="text"
+              valueMultiloc={phaseAttrs.title_multiloc}
+              onChange={handleTitleMultilocOnChange}
             />
-          </PhaseForm>
-        </>
-      );
-    }
-
-    return null;
-  }
-}
-
-const AdminProjectTimelineEditWithHOCs = injectIntl(AdminProjectTimelineEdit);
-
-const Data = adopt<DataProps, InputProps & WithRouterProps>({
-  phases: ({ params, render }) => (
-    <GetPhases projectId={params.projectId}>{render}</GetPhases>
-  ),
-});
-
-export default withRouter((inputProps: InputProps & WithRouterProps) => (
-  <Data {...inputProps}>
-    {(dataProps) => (
-      <AdminProjectTimelineEditWithHOCs {...dataProps} {...inputProps} />
-    )}
-  </Data>
-));
+            <Error apiErrors={errors && errors.title_multiloc} />
+          </SectionField>
+          <SectionField>
+            <ParticipationContext
+              phase={!isNilOrError(phase) ? { data: phase } : undefined}
+              onSubmit={handleParticipationContextOnSubmit}
```

Comment:
Here I ended up needing to add the isNilOrError check inside the component since when we're adding a new phase it will always be null and we want to pass "undefined" so that the participation method selection still shows up.

Reply:
Makes sense! 👍